### ISSUE TITLE
ESLint Update Breaking Plugin?

### ISSUE THREAD

**Issue by leepowellcouk** [mradionov/eslint-plugin-disable#25]
https://github.com/mradionov/eslint-plugin-disable/blob/e66cb5100b3575e820b4033255d4f3e757f39639/index.js#L88

I'm trying to utilise this plugin, however it seems this line is causing the plugin to fail silently. `getConfigForFile` now requires a file path and will error without it, however you're catching the error and silently swallowing it - ESLint does not output this error. 

I'm using ESLint 6.6.0

**Comment by mradionov** [comment]
Thanks, I'll take a look

**Comment by mradionov** [comment]
Notes for myself:

Call to `engine.getConfigForFile()` was introduced in https://github.com/mradionov/eslint-plugin-disable/issues/22 to be able to get root ESLint config with `cliArgs` setting. Overall it was a workaround for `eslint-plugin-vue` and `@vue/cli-service`: the former provides a processor for `.vue` files and the latter uses ESLint Node.js API to lint project files ([file extensions are hardcoded](https://github.com/vuejs/vue-cli/blob/dev/packages/%40vue/cli-plugin-eslint/eslintOptions.js#L27)). There was no way for `disable` plugin to hook in.

In ESLint 6+ there has been changes to processors - https://github.com/eslint/rfcs/tree/master/designs/2018-processors-improvements, which can influence `disable` plugin. `overrides` can probably be used to drop `paths` and `allExceptPaths` settings.

Nevertheless, there is still no clean option to chain processors. You can chain only by changing a file extension and returning a virtual file from `preprocess`. If `preprocess` returns plain text, no more processors will be invoked with this text.

Idea was to use `overrides` to redirect `disable` plugin to `vue` plugin using temporary extension for virtual files:

```
overrides": [
  {
    "files": ["*.vue"],
    "processor": "disable/disable"
  },
  {
    "files": ["*.tmp"],
    "processor": "vue/.vue"
   }
]
```

but unregistered extensions are filtered out:
- [filterCodeBlock call]()https://github.com/eslint/eslint/blob/61749c94bd8a2ebcdfb89e0cd48c4a029a945079/lib/linter/linter.js#L1288
- [filterCodeBlock implementation](https://github.com/eslint/eslint/blob/ca3b2a62c9e829dc4534bca3643d6bc729b46df0/lib/cli-engine/cli-engine.js#L239)

**Comment by oller** [comment]
+1 for this, it would seem this plugin still fails silently when used with eslint 6.  It's a shame as it's the most elegant way to handle these exceptions.

**Comment by mradionov** [comment]
@oller Thanks for the reminder, I've kinda left it off after first investigation because I could not find a solution back then. I'll try to revisit it now.

**Comment by oller** [comment]
no problem, good luck with it! 🤞🏼

**Comment by mradionov** [comment]
I've released a new major version 2.0.0 of the plugin to support ESLint 6+, it has breaking changes.

- [New documentation](https://github.com/mradionov/eslint-plugin-disable/blob/master/README.md)
- [Release info](https://github.com/mradionov/eslint-plugin-disable/releases/tag/v2.0.0)
- [Migration guide](https://github.com/mradionov/eslint-plugin-disable/blob/master/MIGRATING.md)

Please let me know if you have any issues or questions regarding new API.
